src/main/java/com/celements/copydoc/CopyDocumentService.java

```diff
@@ -197,27 +197,29 @@ public boolean copyObject(BaseObject srcObj, BaseObject trgObj) {
 
   boolean copyObject(BaseObject srcObj, BaseObject trgObj, boolean set) {
     boolean hasChanged = false;
-    Set<String> srcProps = srcObj.getPropertyList();
-    Set<String> trgProps = new HashSet<>(trgObj.getPropertyList());
-    for (String name : srcProps) {
-      Object srcVal = modelAccess.getProperty(srcObj, name);
-      Object trgVal = modelAccess.getProperty(trgObj, name);
-      if (!Objects.equal(srcVal, trgVal)) {
+    if (srcObj != trgObj) {
+      Set<String> srcProps = srcObj.getPropertyList();
+      Set<String> trgProps = new HashSet<>(trgObj.getPropertyList());
+      for (String name : srcProps) {
+        Object srcVal = modelAccess.getProperty(srcObj, name);
+        Object trgVal = modelAccess.getProperty(trgObj, name);
+        if (!Objects.equal(srcVal, trgVal)) {
+          if (set) {
+            modelAccess.setProperty(trgObj, name, srcVal);
+          }
+          hasChanged = true;
+          LOGGER.trace("for doc '{}' field '{}' changed from '{}' to '{}'",
+              trgObj.getDocumentReference(), name, trgVal, srcVal);
+        }
+        trgProps.remove(name);
+      }
+      for (String name : trgProps) {
         if (set) {
-          modelAccess.setProperty(trgObj, name, srcVal);
+          trgObj.removeField(name);
         }
         hasChanged = true;
-        LOGGER.trace("for doc '{}' field '{}' changed from '{}' to '{}'",
-            trgObj.getDocumentReference(), name, trgVal, srcVal);
-      }
-      trgProps.remove(name);
-    }
-    for (String name : trgProps) {
-      if (set) {
-        trgObj.removeField(name);
+        LOGGER.trace("for doc '{}' field '{}' set to null", trgObj.getDocumentReference(), name);
       }
-      hasChanged = true;
-      LOGGER.trace("for doc '{}' field '{}' set to null", trgObj.getDocumentReference(), name);
     }
```

Comment:
Looks to me like a possible or even likely issue. Please add logging ob **warn** or **Info** Level.
```suggestion
    } else {
      LOGGER.warn("skip copyObject reason: identical reference");
    }
```